crates/autopilot/src/solvable_orders.rs

```diff
@@ -657,22 +657,22 @@ fn to_normalized_price(price: f64) -> Option<U256> {
 }
 
 async fn filter_unsupported_tokens(
-    mut orders: Vec<Order>,
-    bad_token: &dyn BadTokenDetecting,
+    orders: Vec<Order>,
+    bad_token: Arc<dyn BadTokenDetecting>,
 ) -> Result<Vec<Order>> {
-    // Can't use normal `retain` or `filter` because the bad token detection is
-    // async. So either this manual iteration or conversion to stream.
-    let mut index = 0;
-    'outer: while index < orders.len() {
-        for token in orders[index].data.token_pair().unwrap() {
-            if !bad_token.detect(token).await?.is_good() {
-                orders.swap_remove(index);
-                continue 'outer;
+    try_join_all(orders.into_iter().map(|order| {
```

Comment:
Also we should probably not abort the whole thing if a single error happens and instead not include any orders where we couldn't determine the quality for at least 1 of the traded tokens.